### DISCUSSION TITLE
inverse_reading_order for folder on SD card

### DISCUSSION THREAD

**Discussion by komadorirobin** [koreader/koreader#12151] [Q&A]
* KOReader version: 2024.04
* Device: Boox Go Color 7

I'm trying to add a rule to the settings.reader.lua file to inverse the reading direction to LTR for manga. It doesn't have any effect when adding the rule to an SD card.
 
The storage path I've entered is taken directly from the Koreader file browser.

["/storage/3435-3332/Manga"] = {
["inverse_reading_order"] = true,
["kopt_zoom_direction"] = 0,
},

**Comment by benoit-pierre** [comment]
Is that the whole file?

**Reply by komadorirobin** [reply]
No, that's only the part between the  "-- ADD YOUR DEFAULTS HERE:" and "}--do NOT change this line"

**Reply by benoit-pierre** [reply]
So just to be clear, like this:
```lua
return {
    ["/storage/3435-3332/Manga"] = {
        ["inverse_reading_order"] = true,
        ["kopt_zoom_direction"] = 0,
    },
}
```

**Reply by komadorirobin** [reply]
Yes. I've also tried without the "return", but neither variants work.

This is the entire document:

return {--do NOT change this line

 --HELP:
 -- You can override default settings for documents per directory in this file.
 -- The directories must be under the home folder.
 -- You can find settings to change in the metadata.lua in the .sdr folder.
 -- The path must start with "/" (absolute path).
 -- The path must not end with a "/" it must end with the folder name.

 -- syntax:

 --   ["path/to/folder"] = {
 --         ["setting_to_override"] = value,
 --    },

 -- examples:

 --[[
    ["/mnt/us/documents/hebrew"] = {
        ["inverse_reading_order"] = true
    },
    ["/mnt/onboard/smalltext"] = {
        ["copt_font_size"] = 34,
        ["copt_line_spacing"] = 130,
    },
    ["/sdcard/Books/smalltext"] = {
        ["copt_font_size"] = 34,
        ["copt_line_spacing"] = 130,
    },
--]]

 -- comment out line ("--" at line start) to disable


 -- ADD YOUR DEFAULTS HERE:

 return {
     ["/storage/3435-3332/Manga"] = {
         ["inverse_reading_order"] = true,
         ["kopt_zoom_direction"] = 0,
     },
 }

}--do NOT change this line

**Reply by mergen3107** [reply]
You have TWO nested returns. 
what you need is `return { CONTENT }`
where content is 
```
["/storage/3435-3332/Manga"] = {
["inverse_reading_order"] = true,
["kopt_zoom_direction"] = 0,
},
```

**Reply by benoit-pierre** [reply]
Yep, try my version above, and only that, no other lines.

**Reply by komadorirobin** [reply]
Still doesn't work.

**Reply by benoit-pierre** [reply]
The whole file is exactly like [this](https://github.com/koreader/koreader/discussions/12151#discussioncomment-9997221)?

How are you testing? I'm not sure those will be honored if the document has previously been opened.

**Reply by komadorirobin** [reply]
Yes, the whole document like that. I'm aware of the code not taking effect on previously opened books, so I'm just testing on new ones.

**Reply by benoit-pierre** [reply]
OK, maybe the code is checking with another path (the filemanager might canonicalize / fully resolve it before display). Is the sdcard available under another? Maybe "/sdcard/Manga" directly?

**Reply by komadorirobin** [reply]
No other name from what I can tell. I've just now added "/sdcard/Manga" as well as a few other thinkable names but to no avail.

**Reply by benoit-pierre** [reply]
Try using `adb logcat | grep 'KOReader: opening file'` to double check which directory is used.

**Reply by komadorirobin** [reply]
Just tested and it is indeed "/storage/3435-3332/Manga". This is weird.

**Reply by benoit-pierre** [reply]
This comment in the default template is concerning: ` -- The directories must be under the home folder.`.

**Reply by mergen3107** [reply]
Is this android? What is your device and OS? 
Sometimes it has symlinks to “sdcard”, you probably will have better luck with actual location. 
say, “/storage/emulated/1” or something

also, attach verbose logs

**Reply by Frenzie** [reply]
> This comment in the default template is concerning: ` -- The directories must be under the home folder.`.

Strange, I wonder why they would have to be. It does seem to be the case that it does something with that:

https://github.com/koreader/koreader/blob/bbe4ce466b1a37fc3eb0fa6b38b4bcc791e58784/plugins/docsettingtweak.koplugin/main.lua#L105-L116

**Reply by yparitcher** [reply]
IIRC, when i first made this we decided to only check for folders under one's home directory, so as not to needlessly slow down opening random books, (as the thought was that folders that are commonly used will be somewhere under home). Android file paths were not considered during the creation of the plugin.

**Comment by hius07** [comment]
Real path of the opened file can be taken from koreader/history.lua.
